pkg/sources/source.go

```diff
@@ -200,69 +207,90 @@ func (sp *SourceProcessor) Start(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("failed to create gRPC client, %w", err)
 		}
+
 		// Readiness check
 		if err = udsGRPCClient.WaitUntilReady(ctx); err != nil {
-			return fmt.Errorf("failed on user-defined source readiness check, %w", err)
+			return fmt.Errorf("failed on user-defined sourceReader readiness check, %w", err)
 		}
+
 		defer func() {
 			err = udsGRPCClient.CloseConn(ctx)
 			if err != nil {
 				log.Warnw("Failed to close gRPC client conn", zap.Error(err))
 			}
 		}()
-		readyCheckers = append(readyCheckers, udsGRPCClient)
+		healthCheckers = append(healthCheckers, udsGRPCClient)
 	}
-	maxMessageSize := sharedutil.LookupEnvIntOr(dfv1.EnvGRPCMaxMessageSize, sdkclient.DefaultGRPCMaxMessageSize)
-	var err error
+
 	if sp.VertexInstance.Vertex.HasUDTransformer() {
 		// Wait for server info to be ready
 		serverInfo, err := sdkserverinfo.SDKServerInfo(sdkserverinfo.WithServerInfoFilePath(sdkclient.SourceTransformerServerInfoFile))
 		if err != nil {
 			return err
 		}
 
-		sdkClient, err = sourcetransformer.New(serverInfo, sdkclient.WithMaxMessageSize(maxMessageSize))
+		sdkClient, err := sourcetransformer.New(serverInfo, sdkclient.WithMaxMessageSize(maxMessageSize))
 		if err != nil {
-			return fmt.Errorf("failed to create gRPC client, %w", err)
+			return fmt.Errorf("failed to create transformer gRPC client, %w", err)
 		}
 
-		transformerGRPCClient := transformer.NewGRPCBasedTransformer(sdkClient)
+		transformerClient = transformer.NewGRPCBasedTransformer(sdkClient)
```

Comment:
```suggestion
		srcTransformerGRPCClient = transformer.NewGRPCBasedTransformer(sdkClient)
```

nit: this is to be consistent with srcClient/udsGRPCClient above for udsource.